### PR TITLE
Add chat interop scenario quickstart code

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
Provide chat interoperability quickstart source code to enable use of call with chat scenarios.

Companion code for https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/chat/meeting-interop?pivots=platform-android

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: join-chat-to-teams-meeting/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+1. Add the communications service endpoint URL at line 55:
+   ```private static final String ResourceUrl = "<COMMUNICATION_SERVICES_RESOURCE_ENDPOINT>";```
+
+## Run the sample
+
+Open the sample project using Android Studio and run the application.
+
+![Final look and feel of the quick start app](../Media/join-teams-meeting-chat-quickstart.png)
+
+Insert the Teams meeting URL into the text box, the chat thread ID into the **Meeting Thread Id** text box, and press **Join Meeting** to join the Teams meeting from within your Communication Services application.
```
Comment (vhuseinova-msft) [inline join-chat-to-teams-meeting/README.md:57]: it mentiones chat thread id here and meeting thread id on the screenshot and in the app. Could you update it to be consistent?

File: Media/join-teams-meeting-chat-quickstart.png
Comment (vhuseinova-msft) [inline Media/join-teams-meeting-chat-quickstart.png:1]: Question: Is it possible to remove the line at the bottom of the screen?
image

File: join-chat-to-teams-meeting/README.md (hunk 1 above)
Comment (emlynmac) [reply join-chat-to-teams-meeting/README.md:57]: I think this calls out that they can have different names but are the same thing?
It's important that this is the meeting thread ID as that will be the only one that works, but the APIs to get the chat thread call it the chat thread ID

**Review by vhuseinova-msft — APPROVED** [review]

File: join-chat-to-teams-meeting/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+For full instructions on how to build this code sample from scratch, look at [Quickstart: Join your chat app to a Teams meeting](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/chat/meeting-interop?pivots=platform-android)
+
+## Prerequisites
+
+To complete this tutorial, you’ll need the following prerequisites:
+
+- An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+- [Android Studio](https://developer.android.com/studio), for running your Android application.
+- A deployed Communication Services resource. [Create a Communication Services resource](https://docs.microsoft.com/azure/communication-services/quickstarts/create-communication-resource).
+- A [User Access Token](https://docs.microsoft.com/azure/communication-services/quickstarts/access-tokens) for your Azure Communication Service.
```
Comment (ahammer) [inline join-chat-to-teams-meeting/README.md:21]: A Teams meeting link and Meeting/chat Thread ID?

File: join-chat-to-teams-meeting/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+| CommunicationTokenCredential  | The CommunicationTokenCredential  is used as the token credential to instantiate the CallAgent.|
+
+## Before running sample code
+
+1. Open an instance of PowerShell, Windows Terminal, Command Prompt or equivalent and navigate to the directory that you'd like to clone the sample to.
+1. `git clone https://github.com/Azure-Samples/Communication-Services-Android-Quickstarts.git`
+1. With the `User Token` procured in pre-requisites, add it to the **./app/src/main/java/com/contoso/acsquickstart/MainActivity.java** file. Assign your access token in line 51:
+   ```private static final String UserToken = "<User_Access_Token>";```
+1. Add the created user ID in line 54:
+   ```private static final String InitiatorId = "<USER_ID>";```
```
Comment (ahammer) [inline join-chat-to-teams-meeting/README.md:47]: why not UserId or InitiatingUserId if the word initiator is important here?

File: join-chat-to-teams-meeting/README.md (hunk 2 above)
Comment (emlynmac) [reply join-chat-to-teams-meeting/README.md:21]: The thread ID is generated as part of the Teams meeting. This is wrapped up in the Teams deployment below.

File: join-chat-to-teams-meeting/README.md (hunk 3 above)
Comment (emlynmac) [reply join-chat-to-teams-meeting/README.md:47]: This is from the existing calling quickstart.

File: join-chat-to-teams-meeting/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| CallAgent  | The CallAgent is used to start and manage calls. |
+| ChatThreadAsyncClient | The ChatThreadAsyncClient is the main entry point to the Chat client library. |
+| CommunicationTokenCredential  | The CommunicationTokenCredential  is used as the token credential to instantiate the CallAgent.|
+
+## Before running sample code
+
+1. Open an instance of PowerShell, Windows Terminal, Command Prompt or equivalent and navigate to the directory that you'd like to clone the sample to.
+1. `git clone https://github.com/Azure-Samples/Communication-Services-Android-Quickstarts.git`
+1. With the `User Token` procured in pre-requisites, add it to the **./app/src/main/java/com/contoso/acsquickstart/MainActivity.java** file. Assign your access token in line 51:
+   ```private static final String UserToken = "<User_Access_Token>";```
```
Comment (ahammer) [inline join-chat-to-teams-meeting/README.md:45]: probably should follow better practice for passing/storing the token in.

Can use BUILD_CONFIG_FIELD in gradle, and read a local.properties file (not checked in, added to .gitIgnore). Then pass tokens that way.

It's not good to put tokens that could be checked in, i.e. say someone forks and customizes and pushes this, they could accidentally leak a token.

File: join-chat-to-teams-meeting/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.contoso.acsquickstart" xmlns:tools="http://schemas.android.com/tools">
+
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+    <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (ahammer) [inline join-chat-to-teams-meeting/app/src/main/AndroidManifest.xml:10]: Need?

File: join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,302 @@
+package com.contoso.acsquickstart;
```
Comment (ahammer) [inline join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java:1]: copyright notices

File: join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+        // Register the method for sending messages and toggle the visibility of chat components
+        sendMessageButton.setOnClickListener(l -> sendMessage());
+        sendMessageButton.setVisibility(View.VISIBLE);
+        messageBody.setVisibility(View.VISIBLE);
+
+        // Start the polling for chat messages immediately
+        handler.post(runnable);
+    }
+
+    private Handler handler = new Handler();
```
Comment (ahammer) [inline join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java:129]: Why Java, Kotlin could be much compressed.

But for a repeating timer, 
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html

Also a good idea to check "isFinishing" and "isDestroyed" on the activity before doing something in a thread.

File: join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+    private static final String UserToken = "<User_Access_Token>";
+
+    // InitiatorId is used to differentiate incoming messages from outgoing messages
+    private static final String InitiatorId = "<USER_ID>";
+    private static final String ResourceUrl = "<COMMUNICATION_SERVICES_RESOURCE_ENDPOINT>";
+    private String threadId;
+    private ChatThreadAsyncClient chatThreadAsyncClient;
+
+    // The list of ids corresponsding to messages which have already been processed
+    ArrayList<String> chatMessages = new ArrayList<>();
```
Comment (ahammer) [inline join-chat-to-teams-meeting/app/src/main/java/com/contoso/acsquickstart/MainActivity.java:60]: List<String> if you don't need anything specific to ArrayList()

**Pull request merged**